morf-core/src/main/java/org/alfasoftware/morf/sql/SelectStatementBuilder.java

```diff
@@ -314,13 +314,27 @@ public SelectStatementBuilder allowParallelDml() {
    *
    * @param customHint representation of a custom hint
    * @return this, for method chaining.
+   * @deprecated See {@link org.alfasoftware.morf.sql.CustomHint}
    */
+  @Deprecated
   public org.alfasoftware.morf.sql.SelectStatementBuilder withCustomHint(CustomHint customHint) {
     this.hints.add(customHint);
     return this;
   }
 
 
+  /**
+   * Supplies a specified custom hint and database type to the database for a query.
+   *
+   * @param dialectSpecificHint a custom dialect specific hint
+   * @return this, for method chaining.
+   */
+  public org.alfasoftware.morf.sql.SelectStatementBuilder withDialectSpecificHint(DialectSpecificHint dialectSpecificHint) {
+    this.hints.add(dialectSpecificHint);
+    return this;
+  }

```

Comment:
This should have `String databaseType, String hintContents` parameters.

(SelectStatementBuilder and SelectStatement method signatures should be identical and interchangeable)
(I know I suggested otherwise during our pair-programming, I should have pointed this out later)

Reply:
Done. Waiting to fix the other comments before pushing all the changes.